Support for OpenBSD

Please provide support for OpenBSD.

> OpenBSD is **NOT** supported, and you definitely did **not** check before closing this issue.
> 
> Building from source fails:
> 
> ```shell
> $ cargo build
>   Downloaded libc v0.2.133
>   Downloaded rand_core v0.6.4
>   Downloaded syn v1.0.100
>   Downloaded thiserror-impl v1.0.35
>   Downloaded thiserror v1.0.35
>   Downloaded unicode-ident v1.0.4
>   Downloaded unicode-normalization v0.1.22
>   Downloaded unicode-segmentation v1.10.0
>   Downloaded cap-time-ext v0.25.3
>   Downloaded open v3.0.3
>   Downloaded cap-fs-ext v0.25.3
>   Downloaded iana-time-zone v0.1.48
>   Downloaded rgb v0.8.34
>   Downloaded exr v1.5.1
>   Downloaded unicode-width v0.1.10
>   Downloaded lock_api v0.4.9
>   Downloaded once_cell v1.15.0
>   Downloaded unicode-script v0.5.5
>   Downloaded pest v2.3.1
>   Downloaded textwrap v0.15.1
>   Downloaded itertools v0.10.5
>   Downloaded reqwest v0.11.12
>   Downloaded clap v3.2.22
>   Downloaded env_logger v0.9.1
>   Downloaded cap-rand v0.25.3
>   Downloaded anyhow v1.0.65
>   Downloaded cap-primitives v0.25.3
>   Downloaded jsonrpc-lite v0.6.0
>   Downloaded cap-std v0.25.3
>   Downloaded tokio v1.21.1
>   Downloaded tree-sitter-swift v0.3.3
>   Downloaded 31 crates (3.7 MB) in 2.77s
>    Compiling libc v0.2.133
>    Compiling proc-macro2 v1.0.43
>    Compiling quote v1.0.21
>    Compiling unicode-ident v1.0.4
>    Compiling syn v1.0.100
>    Compiling autocfg v1.1.0
>    Compiling cfg-if v1.0.0
>    Compiling memchr v2.5.0
>    Compiling thiserror v1.0.35
>    Compiling smallvec v1.9.0
>    Compiling pkg-config v0.3.25
>    Compiling once_cell v1.15.0
>    Compiling serde_derive v1.0.144
>    Compiling anyhow v1.0.65
>    Compiling bitflags v1.3.2
>    Compiling serde v1.0.144
>    Compiling version_check v0.9.4
>    Compiling regex-syntax v0.6.27
>    Compiling log v0.4.17
>    Compiling unicode-segmentation v1.10.0
>    Compiling either v1.8.0
>    Compiling strum v0.21.0
>    Compiling version-compare v0.0.11
>    Compiling scopeguard v1.1.0
>    Compiling hashbrown v0.12.3
>    Compiling itoa v1.0.3
>    Compiling futures-core v0.3.24
>    Compiling pin-project-lite v0.2.9
>    Compiling adler v1.0.2
>    Compiling crc32fast v1.3.2
>    Compiling lazy_static v1.4.0
>    Compiling futures-sink v0.3.24
>    Compiling crossbeam-utils v0.8.11
>    Compiling stable_deref_trait v1.2.0
>    Compiling fallible-iterator v0.2.0
>    Compiling futures-io v0.3.24
>    Compiling futures-task v0.3.24
>    Compiling futures-channel v0.3.24
>    Compiling io-lifetimes v0.7.3
>    Compiling byteorder v1.4.3
>    Compiling futures-util v0.3.24
>    Compiling pin-utils v0.1.0
>    Compiling rustix v0.35.9
>    Compiling typenum v1.15.0
>    Compiling ucd-trie v0.1.5
>    Compiling rayon-core v1.9.3
>    Compiling unicode-bidi v0.3.8
>    Compiling target-lexicon v0.12.4
>    Compiling cranelift-isle v0.86.1
>    Compiling base64 v0.13.0
>    Compiling rustc-demangle v0.1.21
>    Compiling fnv v1.0.7
>    Compiling unicode-width v0.1.10
>    Compiling cranelift-codegen-shared v0.86.1
>    Compiling ryu v1.0.11
>    Compiling bytes v1.2.1
>    Compiling leb128 v0.2.5
>    Compiling parking_lot_core v0.9.3
>    Compiling slice-group-by v0.3.0
>    Compiling tinyvec_macros v0.1.0
>    Compiling arrayvec v0.5.2
>    Compiling cpufeatures v0.2.5
>    Compiling ppv-lite86 v0.2.16
>    Compiling opaque-debug v0.3.0
>    Compiling io-extras v0.15.0
>    Compiling heck v0.4.0
>    Compiling percent-encoding v2.2.0
>    Compiling ambient-authority v0.0.1
>    Compiling more-asserts v0.2.2
>    Compiling cap-primitives v0.25.3
>    Compiling ipnet v2.5.0
>    Compiling serde_json v1.0.85
>    Compiling zstd-safe v5.0.2+zstd.1.5.2
>    Compiling termcolor v1.1.3
>    Compiling same-file v1.0.6
>    Compiling maybe-owned v0.3.4
>    Compiling humantime v2.1.0
>    Compiling cpp_demangle v0.3.5
>    Compiling wasmtime-cache v0.39.1
>    Compiling xmlparser v0.13.3
>    Compiling async-trait v0.1.57
>    Compiling matches v0.1.9
>    Compiling cap-std v0.25.3
>    Compiling openssl v0.10.41
>    Compiling unicode-script v0.5.5
>    Compiling linked-hash-map v0.5.6
>    Compiling wasmtime v0.39.1
>    Compiling unicode-ccc v0.1.2
>    Compiling arrayvec v0.7.2
>    Compiling unicode-general-category v0.4.0
>    Compiling unicode-bidi-mirroring v0.1.0
>    Compiling parking_lot_core v0.8.5
>    Compiling fixedbitset v0.2.0
>    Compiling foreign-types-shared v0.1.1
>    Compiling tinystr v0.3.4
>    Compiling gio v0.14.8
>    Compiling cfg-if v0.1.10
>    Compiling const-cstr v0.3.0
>    Compiling httparse v1.8.0
>    Compiling native-tls v0.2.10
>    Compiling openssl-probe v0.1.5
>    Compiling unic-common v0.9.0
>    Compiling font-kit v0.11.0 (https://github.com/lapce/font-kit#23e28b06)
>    Compiling weezl v0.1.7
>    Compiling ttf-parser v0.15.2
>    Compiling wasi-common v0.39.1
>    Compiling paste v1.0.9
>    Compiling unic-char-range v0.9.0
>    Compiling proc-macro-hack v0.5.19
>    Compiling encoding_rs v0.8.31
>    Compiling pico-args v0.4.2
>    Compiling strsim v0.10.0
>    Compiling cap-fs-ext v0.25.3
>    Compiling unicode-vo v0.1.0
>    Compiling try-lock v0.2.3
>    Compiling siphasher v0.3.10
>    Compiling color_quant v1.1.0
>    Compiling yazi v0.1.4
>    Compiling float-cmp v0.9.0
>    Compiling zeno v0.2.2
>    Compiling system-interface v0.21.0
>    Compiling xmlwriter v0.1.0
>    Compiling float-ord v0.2.0
>    Compiling unicode-general-category v0.2.0
>    Compiling half v1.8.2
>    Compiling bit_field v0.10.1
>    Compiling cache-padded v1.2.0
>    Compiling signal-hook v0.3.14
>    Compiling rctree v0.4.0
>    Compiling arrayref v0.3.6
>    Compiling tower-service v0.3.2
>    Compiling httpdate v1.0.2
>    Compiling bytecount v0.6.3
>    Compiling lebe v0.5.2
>    Compiling rustversion v1.0.9
>    Compiling scoped_threadpool v0.1.9
>    Compiling fastrand v1.8.0
>    Compiling ident_case v1.0.1
>    Compiling waker-fn v1.1.0
>    Compiling rustc-hash v1.1.0
>    Compiling wasmtime-wasi v0.39.1
>    Compiling event-listener v2.5.3
>    Compiling parking v2.0.0
>    Compiling glob v0.3.0
>    Compiling lexical-core v0.7.6
>    Compiling num_threads v0.1.6
>    Compiling atomic-waker v1.0.0
>    Compiling iana-time-zone v0.1.48
>    Compiling ttf-parser v0.9.0
>    Compiling async-task v4.3.0
>    Compiling float-cmp v0.5.3
>    Compiling utf8parse v0.2.0
>    Compiling lazycell v1.3.0
>    Compiling os_str_bytes v6.3.0
>    Compiling druid-shell v0.7.0 (https://github.com/lapce/druid?branch=shell_opengl#ae9a85d2)
>    Compiling static_assertions v1.1.0
>    Compiling ttf-parser v0.12.3
>    Compiling glow v0.11.2
>    Compiling mime v0.3.16
>    Compiling siphasher v0.2.3
>    Compiling textwrap v0.15.1
>    Compiling self_cell v0.10.2
>    Compiling intmap v0.7.1
>    Compiling serde v0.8.23
>    Compiling arc-swap v1.5.1
>    Compiling pulldown-cmark v0.9.2
>    Compiling strum v0.24.1
>    Compiling rctree v0.3.3
>    Compiling ansi_term v0.12.1
>    Compiling process_path v0.1.4
>    Compiling rust-ini v0.13.0
> error[E0433]: failed to resolve: use of undeclared crate or module `os`
>   --> /home/koutheir/.cargo/registry/src/github.com-1ecc6299db9ec823/process_path-0.1.4/src/lib.rs:58:5
>    |
> 58 |     os::get_executable_path()
>    |     ^^ use of undeclared crate or module `os`
> 
> error[E0433]: failed to resolve: use of undeclared crate or module `nix`
>   --> /home/koutheir/.cargo/registry/src/github.com-1ecc6299db9ec823/process_path-0.1.4/src/lib.rs:71:9
>    |
> 71 |         nix::get_dylib_path()
>    |         ^^^ use of undeclared crate or module `nix`
> 
> For more information about this error, try `rustc --explain E0433`.
> error: could not compile `process_path` due to 2 previous errors
> warning: build failed, waiting for other jobs to finish...
> ```

Lapce is not the issue, it is the fact that Rust is not fully supported on OpenBSD.

--- a/lapce-data/Cargo.toml
+++ b/lapce-data/Cargo.toml
@@ -10,7 +10,6 @@ zip = { version = "0.6.2", default-features = false, features = ["deflate"] }
 tar = "0.4.38"
 flate2 = "1.0.24"
 fs_extra = "1.2.0"
-process_path = "0.1.4"
 url = "2.2.2"
 bitflags = "1"
 chrono = "0.4.19"

--- a/lapce-data/src/data.rs
+++ b/lapce-data/src/data.rs
@@ -282,7 +282,7 @@ impl LapceData {
     }
 
     fn listen_local_socket(event_sink: ExtEventSink) -> Result<()> {
-        if let Some(path) = process_path::get_executable_path() {
+        if let Ok(path) = std::env::current_exe() {
             if let Some(path) = path.parent() {
                 if let Some(path) = path.to_str() {
                     if let Ok(current_path) = std::env::var("PATH") {
@@ -1138,9 +1138,7 @@ impl LapceTabData {
             LapceWorkbenchCommand::RestartToUpdate => {
                 if let Some(release) = (*self.latest_release).clone() {
                     if release.version != *VERSION {
-                        if let Some(process_path) =
-                            process_path::get_executable_path()
-                        {
+                        if let Ok(process_path) = std::env::current_exe() {
                             ctx.submit_command(Command::new(
                                 LAPCE_UI_COMMAND,
                                 LapceUICommand::RestartToUpdate(

--- a/lapce-proxy/Cargo.toml
+++ b/lapce-proxy/Cargo.toml
@@ -7,7 +7,6 @@ edition = "2021"
 [dependencies]
 interprocess = "1.1.1"
 clap = { version = "3.2.17", features = ["derive"] }
-process_path = "0.1.4"
 once_cell = "1.15"
 url = "2.2.2"
 dyn-clone = "1.0.8"

--- a/lapce-proxy/src/lib.rs
+++ b/lapce-proxy/src/lib.rs
@@ -130,7 +130,7 @@ pub fn mainloop() {
     std::thread::spawn(move || {
         let _ = listen_local_socket(local_proxy_rpc);
     });
-    if let Some(path) = process_path::get_executable_path() {
+    if let Ok(path) = std::env::current_exe() {
         if let Some(path) = path.parent() {
             if let Some(path) = path.to_str() {
                 if let Ok(current_path) = std::env::var("PATH") {

